Refactor for more features

ghostflyby opened the pull request:
This is a draft on refactoring for some new features.
1. more configurable options
2. functioning in AE terminals
3. easily adding new types of overlays

ghostflyby committed:
Do not force my JVM vendor

ghostflyby committed:
Refactor: extract inner static class

ghostflyby committed:
New: DurabilityLikeFormatter util

ghostflyby committed:
Refactor: DurabilityFormatter util uses `String.format`

ghostflyby committed:
Refactor: DurabilityRenderer

ghostflyby committed:
New: ItemHandler interface in its own file

ghostflyby committed:
Spotless Apply

ghostflyby committed:
Refactor: config into multiple files

ghostflyby committed:
Refactor: config into multiple files :spotlessApply

ghostflyby committed:
Refactor: Moving handlers to multiple files

ghostflyby committed:
Refactor: spotlessApply

ghostflyby committed:
Fix: handleDefault logic

ghostflyby committed:
Fix: handleDefault logic

ghostflyby committed:
Refactor: move configs inside Overlays

ghostflyby committed:
Refactor: move configs inside Overlays: spotlessApply

ghostflyby committed:
Fix: show Overlay for mouse picked item

ghostflyby committed:
Fix: functioning in AE2 terminals

ghostflyby committed:
Refactor: construct Overlay from durability

ghostflyby committed:
Merged changes

Caedis commented on src/main/java/com/caedis/duradisplay/mixins/minecraft/MixinAppEngRenderItem.java:
Our AE2 fork has an API you can call instead of using a mixin (should be used since we have full control over this mod)

Added in https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/commit/46233990e464332bdfef18c4057d4ffdbd34a2bd

ghostflyby replied:
The api is currently not working properly, the item count and durability bar visibilities are forced to sync.
<img width="933" alt="image" src="https://github.com/GTNewHorizons/DuraDisplay/assets/57822596/39bd6242-9fbc-4a96-8ca3-4587697c2335">

Caedis replied:
What I started on was basically reimplementing what AppEngItemRenderer was doing but with our modifications.

Let @firenoo know if you want any other changes.

ghostflyby replied:
[6251567](https://github.com/GTNewHorizons/DuraDisplay/pull/9/commits/6251567ce633e01288fe617e22b1e4c113cfb404)

Caedis commented:
Looks great so far. Thanks for the help!

ghostflyby committed:
Update build.gradle

ghostflyby committed:
No zLevel in rendering

ghostflyby committed:
Revert "Update build.gradle"

This reverts commit e4870b18b8fc9cab7e9d4a58768d5255c7a75cd0.

ghostflyby committed:
NumpadPosition Name change

ghostflyby committed:
Refactor: Config and logical structure
Configs are now auto registered using annotation (by extending Config)
OverLays now generates Renderers with input ItemStack
Renderers have different styles (currently only NumPad)

ghostflyby committed:
Renames and clean unused methods

ghostflyby committed:
Merge branch 'refactor_for_ae2cell_support' into refactor

ghostflyby committed:
build script update

ghostflyby committed:
fix sourcesJar task error in CI

ghostflyby committed:
Fix idea not finding class in generated sources

ghostflyby committed:
ignore mixin warnings (safely) with annotation

ghostflyby committed:
New: util for changing mod self-drawn bar

ghostflyby committed:
New: util for changing mod self-drawn bar

ghostflyby committed:
New: Overlay auto load using annotation

ghostflyby committed:
Code cleaning

ghostflyby committed:
Code cleaning: spotless

ghostflyby committed:
Multi-level Config

ghostflyby committed:
New: Enum Config load util

ghostflyby committed:
New: Enum based ColorType

ghostflyby committed:
Fix: text format follows config

ghostflyby committed:
New: basic BarRenderer

ghostflyby committed:
New: addHandler by className, Typo

ghostflyby committed:
Refactor: remove config annotation

ghostflyby committed:
Refactor: SImplify Config

ghostflyby committed:
Refactor: Simplify Config

ghostflyby committed:
Remove unused classes

ghostflyby committed:
Refactor: integration AE2 using api instead of mixin

ghostflyby committed:
Refactor: move DurabilityLikeInfo out

Caedis commented on src/main/java/com/caedis/duradisplay/config/DuraDisplayConfig.java:
```diff
@@ -46,107 +45,20 @@ public static void reloadConfigObject() {
-        GT_Mod.gregtechproxy.mRenderItemDurabilityBar = Enable
-            && !(DurabilityConfig.Enabled && !DurabilityConfig.RenderBar);
-        GT_Mod.gregtechproxy.mRenderItemChargeBar = Enable && !(ChargeConfig.Enabled && !ChargeConfig.RenderBar);
+        GT_Mod.gregtechproxy.mRenderItemDurabilityBar = false;
```
I see you fell for the refactor suggestion.

This logic is actually needed as otherwise the default bars for these 2 mods would never show.

ghostflyby replied:
[95e6ca3](https://github.com/GTNewHorizons/DuraDisplay/pull/9/commits/95e6ca311e0cffc6ed781ae0f076723735d4c904)

ghostflyby committed:
Dependencies are now all optional

ghostflyby committed:
Depends on mods implementation

ghostflyby committed:
Config Mod Self-drawn bars

ghostflyby committed:
Code cleanup

ghostflyby committed:
Code cleanup

ghostflyby committed:
Make contractor public

Caedis commented:
Where can I find `OverlayInfo` or are you still working on that?

ghostflyby commented:
> Where can I find `OverlayInfo` or are you still working on that?

Just build the project once (`./gradlew build` for example), it will be generated.

Caedis commented:
Got it, thanks

Caedis committed:
Shorten numbers

Caedis committed:
Fix: Tinker Tool Durability + Botania Brew

ghostflyby committed:
Change disable behavior

ghostflyby committed:
Show Chargebar as bar

ghostflyby commented:
Option to show bars as bars
<img width="189" alt="image" src="https://github.com/GTNewHorizons/DuraDisplay/assets/57822596/9b785e15-9c02-4927-bb02-7f490fb3b322">

ghostflyby committed:
Seperated botania brew overlay

ghostflyby committed:
Seperated botania brew overlay: applySpotless

ghostflyby committed:
New Overlay for GT lighters and sprays
Known issue: max durability cannot be got

ghostflyby committed:
Charge showWhenFull = true by default

ghostflyby committed:
Code cleanup

ghostflyby committed:
Fix: GT sprays and lighters with correct max

ghostflyby committed:
Renaming and Moving, handlers are public now

ghostflyby committed:
Overlay abstract class method change and comments

ghostflyby committed:
Add descriptions for categories

ghostflyby committed:
Configurable ColorThreshold

ghostflyby committed:
Fix: format double 1.0 + 2.0 != 3.0 problem

ghostflyby committed:
Filtering with UnlocalizedName

ghostflyby committed:
New color type: smooth

like Threshold but smooth

ghostflyby committed:
Fix: getMaxDamage logic

ghostflyby committed:
More items in Gadget category

ghostflyby committed:
Fix: ToolCore Charge

ghostflyby committed:
Fix GT empty and full lighters and sprays

ghostflyby committed:
Remove unnecessary initialization

ghostflyby committed:
Fix AE2 ToolNetherQuartzCuttingKnife

Caedis commented:
Oh, didnt notice you marked this for review.

Are you on the GTNH discord so I can give you the proper credit when I make a post in the upcoming-features channel?

ghostflyby commented:
My id on discord is ghostflyby.

Caedis commented:
Are you working on the requested changes?
If not, I am going to merge tomorrow (2023-09-15)

ghostflyby commented:
Merge it at your will.
